engine/worker/cmd_run.go

```diff
@@ -0,0 +1,360 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/google/gops/agent"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+)
+
+func cmdRun(w *currentWorker) *cobra.Command {
+	c := &cobra.Command{
+		Use:   "run",
+		Short: "worker run.",
+		Long:  "worker run: you should not need this command directly",
+		Run:   runCmd(w),
+	}
+
+	initFlagsRun(c)
```

Comment:
why duplicating the flags and not used persistent flags of the root command ?

Reply:
The persistent flags are persistent again. But flags

```
single-use
auto-update
force-exit
basedir
ttl
booked-pb-job-id
booked-workflow-job-id
booked-job-id
grpc-api
grpc-insecure
graylog-protocol
graylog-host
graylog-port
graylog-extra-key
graylog-extra-value
```
are not useful to other commands than the root/run commands.

Duplicate theses flags between root cmd and run cmd is usefull to run sub-process with

```golang
current, errExec := os.Executable()
args := []string{"run"}
args = append(args, os.Args[1:]...)
cmd := exec.Command(current, args...)
```

Reply:
ok, got it